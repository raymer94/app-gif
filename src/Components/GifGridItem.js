import React from 'react';
import PropTypes from 'prop-types';

const GifGridItem =  ({img: {title, url}}) =>{
    return (
         <div className="card animate__animated animate__bounce">
            <img src={url} alt={title}></img>
            <p>{title}</p>
        </div>
    )
}

export default GifGridItem

GifGridItem.propTypes = {
    img: PropTypes.shape({
        title: PropTypes.string.isRequired,
        url: PropTypes.string.isRequired
    })   
}
