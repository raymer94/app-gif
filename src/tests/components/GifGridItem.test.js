import React from 'react';
import { shallow } from "enzyme"
import GifGridItem from "../../Components/GifGridItem";



describe("Pruebas en <GifGridItem/>",()=>{

    const obj = {title: 'test', url: 'ddd'}    
    const wrapper = shallow(<GifGridItem img={obj}/>);

    test("debe mostrar el componente", () => {         
        expect(wrapper).toMatchSnapshot();
    });

    test("debe mostrar un parraro con el title", () => {
        const parrafo = wrapper.find('p');
        expect(parrafo.text().trim()).toBe(obj.title)
    });

    test("debe tener img igual al url y alt de los props", ()=>{
        const img = wrapper.find('img');
        expect(img.props().src).toBe(obj.url)
    });

    test("evaluando clase de css", () => {
       const div = wrapper.find('div');
       expect(div.props().className.includes("animate__bounce")).toBe(true);
       //console.log(div.props().className)
    });
})