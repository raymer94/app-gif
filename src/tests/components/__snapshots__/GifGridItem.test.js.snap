// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> debe mostrar el componente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="test"
    src="ddd"
  />
  <p>
    test
  </p>
</div>
`;
